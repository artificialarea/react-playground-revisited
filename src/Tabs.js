import React from 'react';



export default class Tabs extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      contentIndex: 0,
      touched: false,
    }
  }

  handleClickHere = (contentIndex) => {
    this.setState({
      contentIndex,
      touched: true
    })
  }

  render() {

    const tabs = this.props.tabs.map((tab, index) => 
      <button 
        key={index}
        onClick={() => this.handleClickHere(index)}
      >
        {tab.name}
      </button>
    );

    let contentText = this.props.tabs[this.state.contentIndex];

    return (
      <div className="tabs">
        {tabs}
        {this.state.touched && <p className="content">{contentText.content}</p>}
      </div>
    )
  }
}

Tabs.defaultProps = {
  tabs: []
}