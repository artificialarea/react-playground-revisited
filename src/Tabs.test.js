import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'; // NOTE: requires configuration set up in setupTests.js
import Tabs from './Tabs';
import tabstore from './tabstore';

describe('<Tabs />', () => {

  // mimicing state from App.js
  const tabsProp = [ ...tabstore ] 

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Tabs />, div)
    ReactDOM.unmountComponentAtNode(div)
  });

  // Unfortunately, can't sufficiently test event listeners with 'react-test-renderer'. Need to use 'Enzyme' testing library.
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<Tabs tabs={tabsProp}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // ENZYME testing of event handlers
  // NOTE: requires configuration set up in setupTests.js
  it('renders empty given no tabs', () => {

  })

});