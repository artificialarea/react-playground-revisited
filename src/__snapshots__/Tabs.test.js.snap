// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> renders UI as expected 1`] = `
<div
  className="tabs"
>
  <button
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    onClick={[Function]}
  >
    Third tab
  </button>
</div>
`;
